packages/signers/README.md

```diff
@@ -0,0 +1,281 @@
+assertIsTransactionSenderSigner({ address: myAddress }); // ❌ Throws an error.
+```
+
+## Creating and generating KeyPair signers
+
+### Types
+
+#### `KeyPairSigner<TAddress>`
+
+Defines a signer that uses a Crypto KeyPair to sign messages and transactions. It implements both the `MessageSigner` and `TransactionSigner` interfaces and keeps track of the `CryptoKeyPair` instance used to sign messages and transactions.
+
+```ts
+import { generateKeyPairSigner } from '@solana/signers';
+
+const myKeyPairSigner = generateKeyPairSigner();
+myKeyPairSigner.address; // Base58EncodedAddress;
+myKeyPairSigner.keyPair; // CryptoKeyPair;
+const [myMessageResponse] = await myKeyPairSigner.signMessage([myMessage]);
+const [mySignedTransaction] = await myKeyPairSigner.signTransaction([myTransaction]);
+```
+
+### Functions
+
+#### `createSignerFromKeyPair()`
+
+Creates a `KeyPairSigner` from a provided Crypto KeyPair. The `signMessage` and `signTransaction` functions of the returned signer will use the private key of the provided key pair to sign messages and transactions. Note that both the `signMessage` and `signTransaction` implementations are parallelized, meaning that they will sign all provided messages and transactions in parallel.
+
+```ts
+import { generateKeyPair } from '@solana/keys';
+import { createSignerFromKeyPair, KeyPairSigner } from '@solana/signers';
+
+const myKeyPair: CryptoKeyPair = await generateKeyPair();
+const myKeyPairSigner: KeyPairSigner = await createSignerFromKeyPair(myKeyPair);
+```
+
+#### `generateKeyPairSigner()`
+
+A convenience function that generates a new Crypto KeyPair and immediately creates a `KeyPairSigner` from it.
+
+```ts
+import { generateKeyPairSigner, KeyPairSigner } from '@solana/signers';
+
+const myKeyPairSigner: KeyPairSigner = await generateKeyPairSigner();
+```
+
+#### `isKeyPairSigner()`
+
+A type guard that returns `true` if the provided value is a `KeyPairSigner`.
+
+```ts
+const myKeyPairSigner = await generateKeyPairSigner();
+isKeyPairSigner(myKeyPairSigner); // ✅ true
+isKeyPairSigner({ address: address('1234..5678') }); // ❌ false
+```
+
+#### `assertIsKeyPairSigner()`
+
+A type guard that throws an error if the provided value is not a `KeyPairSigner`.
+
+```ts
+const myKeyPairSigner = await generateKeyPairSigner();
+assertIsKeyPairSigner(myKeyPairSigner); // ✅ void
+assertIsKeyPairSigner({ address: address('1234..5678') }); // ❌ Throws an error.
+```
+
+## Creating Noop signers
+
+### Functions
+
+#### `createNoopSigner()`
+
+_Coming soon..._
+
+Creates a Noop (No-Operation) signer from a given address. It will return an implementation of both the `MessageSigner` and `TransactionSigner` interfaces that do not sign anything. Namely, signing a transaction will return the transaction as-is and signing a message will provide empty byte arrays as signatures. This is mostly useful for testing purposes.
```

Comment:
Do we have a testing use case for this currently? Or has it been useful in Kinobi etc? I'd be tempted to just skip this and let people define it themselves in tests if they need it.

Reply:
I don't mind not including this out-of-the-box. There are two main use-cases I've seen for noop signers (as people used them in Umi):
- For testing purposes
- In production, to create transactions on the client that require a server signature. Since the generated instruction function requires a signer, you pass a noop signer to make it happy and then you send it over to your server (serialised) to it can sign it for real before sending it.

Reply:
Ah got you, this is `guestIdentityDriver` or something like that in the Metaplex SDK? Definitely found that useful for partial signing! Since we have tree shaking I can't see the harm in adding it.